service/controller/resource/app/current.go

```diff
@@ -33,10 +34,12 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) ([]*v1a
 		r.logger.Debugf(ctx, "finding apps for tenant cluster %#q", key.ClusterID(&cr))
 
 		o := metav1.ListOptions{
-			LabelSelector: fmt.Sprintf("%s=%s", label.ManagedBy, project.Name()),
+			LabelSelector: fmt.Sprintf("%s=%s,%s=%s", label.Cluster, key.ClusterID(&cr), label.ManagedBy, project.Name()),
 		}
 
-		list, err := r.g8sClient.ApplicationV1alpha1().Apps(key.ClusterID(&cr)).List(ctx, o)
+		list := &v1alpha1.AppList{}
+
+		err := r.ctrlClient.List(ctx, list, &client.ListOptions{Raw: &o})
```

Comment:
we are now listing all namespaces, right? wouldn't be better to limit to a single namespace?

Reply:
Good catch 🚀, we can limit a bit, fix https://github.com/giantswarm/cluster-operator/pull/1518